azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -957,6 +999,22 @@ func flattenKubernetesClusterAgentPoolProfiles(profiles *[]containerservice.Mana
 			agentPoolProfile["count"] = int(*profile.Count)
 		}
 
+		if profile.MinCount != nil {
+			agentPoolProfile["min_count"] = int(*profile.MinCount)
+		}
+
+		if profile.MaxCount != nil {
+			agentPoolProfile["max_count"] = int(*profile.MaxCount)
+		}
+
+		if profile.EnableAutoScaling != nil {
+			agentPoolProfile["enable_auto_scaling"] = *profile.EnableAutoScaling
+		}
+
+		if profile.AvailabilityZones != nil && len(*profile.AvailabilityZones) > 0 {
+			agentPoolProfile["availability_zones"] = utils.FlattenStringSlice(profile.AvailabilityZones)
+		}
```

Comment:
can we remove this if statement around this flatten? the `utils.FlattenStringSlice` should handle this being nil (and we should always set this complex type if there's no elements, since that'll then show a diff)

Reply:
@tombuildsstuff:
In this case,  I am protecting us against some behaviours of the API where an empty array is not the same as not sending an array.

Reply:
But here we are setting the terraform side. Any protection on the API side should be where we set that property on the SDK (only set it if tf data is !nil and len > 0)

Reply:
I removed it here and just left it on the expand. Is that OK?